.github/workflows/ci-build-and-push.yml

```diff
@@ -0,0 +1,45 @@
+name: ci-build-and-push
+
+on:
+  push:
+    branches:
+      - main  
+  pull_request:
```

Comment:
we don't want to push on pull request

Reply:
I wasn't sure about it; thanks for the clarification!